src/qtt/algorithms/fitting.py

```diff
@@ -249,6 +250,25 @@ def gaussian_model(x, mean, sigma, amplitude):  # type: ignore
 
     return result_dict['fitted_parameters'], result_dict
 
+def fit_sine(x_data: np.ndarray, y_data: np.ndarray, initial_parameters = None) -> Tuple[Dict[str, Any],Dict[str, Any]]:
+    """ Fit a sine wave for the inputted data; see sine function in functions.py for model
+    """
+    if initial_parameters is None:
+        initial_parameters = _estimate_initial_parameters_sine(x_data, y_data)
+
+    lmfit_model = Model(sine)
+    lmfit_result = lmfit_model.fit(y_data,x=x_data,**dict(zip(lmfit_model.param_names, initial_parameters)))
+    result_dict = extract_lmfit_parameters(lmfit_model, lmfit_result)
+
+    return result_dict['fitted_parameters'], result_dict
+
+def _estimate_initial_parameters_sine(x_data: np.ndarray, y_data: np.ndarray) -> np.ndarray:
+    amplitude = np.max(y_data)-np.min(y_data)
```

Comment:
This is strange: the amplitude feeded to the sine in the test is 2 and here I get 4.08919665781203. How come?
Here actually you are calculating 2 times the amplitude, not the amplitude.